src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -0,0 +1,445 @@
+      _GenerateICallTable;
+      _WasmCompileNativeFiles;
+      _WasmLinkDotNet;
+      _CompleteWasmBuildNative
+    </_WasmBuildNativeCoreDependsOn>
+
+    <WasmBuildNativeOnlyDependsOn>
+      _PrepareForWasmBuildNativeOnly;
+      _WasmBuildNativeCore;
+    </WasmBuildNativeOnlyDependsOn>
+
+    <_BeforeWasmBuildAppDependsOn>
+      $(_BeforeWasmBuildAppDependsOn);
+      _SetupEmscripten;
+      _SetWasmBuildNativeDefaults
+    </_BeforeWasmBuildAppDependsOn>
+
+    <_ExeExt Condition="$([MSBuild]::IsOSPlatform('WINDOWS'))">.exe</_ExeExt>
+    <WasmUseEMSDK_PATH Condition="'$(WasmUseEMSDK_PATH)' == '' and '$(EMSDK_PATH)' != '' and Exists('$(MSBuildThisFileDirectory)WasmApp.InTree.targets')">true</WasmUseEMSDK_PATH>
+
+    <MicrosoftNetCoreAppRuntimePackDir Condition="'$(MicrosoftNetCoreAppRuntimePackDir)' == ''">$([MSBuild]::NormalizeDirectory($(NuGetPackageRoot), 'microsoft.netcore.app.runtime.mono.browser-wasm', '$(BundledNETCoreAppPackageVersion)'))</MicrosoftNetCoreAppRuntimePackDir>
+    <MicrosoftNetCoreAppRuntimePackRidDir Condition="'$(MicrosoftNetCoreAppRuntimePackRidDir)' == ''">$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackDir), 'runtimes', 'browser-wasm'))</MicrosoftNetCoreAppRuntimePackRidDir>
+    <MicrosoftNetCoreAppRuntimePackRidDir>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidDir)))</MicrosoftNetCoreAppRuntimePackRidDir>
+    <MicrosoftNetCoreAppRuntimePackRidNativeDir>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidDir), 'native'))</MicrosoftNetCoreAppRuntimePackRidNativeDir>
+
+    <_WasmRuntimePackIncludeDir>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'include'))</_WasmRuntimePackIncludeDir>
+    <_WasmRuntimePackSrcDir>$([MSBuild]::NormalizeDirectory($(MicrosoftNetCoreAppRuntimePackRidNativeDir), 'src'))</_WasmRuntimePackSrcDir>
+    <_EmccPropsPath>$(_WasmRuntimePackSrcDir)Emcc.props</_EmccPropsPath>
+  </PropertyGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)EmSdkRepo.Defaults.props" Condition="'$(WasmUseEMSDK_PATH)' == 'true'" />
+
+  <!-- Always import it. $(WasmBuildNative) can change when running the targets, eg. when native
+       references are present but the default value for WasmBuildNative was false.
+       So, at evaluation time, it will be false, and we'll end up skipping this import. Instead, always
+       import it. -->
+  <Import Project="$(_EmccPropsPath)" />
+
+  <!-- "public" target meant for use outside the regular wasm app generation process FIXME: rename please! -->
+  <Target Name="WasmBuildNativeOnly" DependsOnTargets="$(WasmBuildNativeOnlyDependsOn)" Condition="'$(WasmBuildNative)' == 'true'" />
+
+  <Target Name="_PrepareForWasmBuildNativeOnly">
+    <ItemGroup>
+      <_WasmAssembliesInternal Remove="@(_WasmAssembliesInternal)" />
+      <_WasmAssembliesInternal Include="@(WasmAssembliesToBundle->Distinct())" />
+    </ItemGroup>
+  </Target>
+
+
+  <Target Name="_SetupEmscripten">
```

Comment:
Should we move the base emscripten detection to the WorkloadManifest.targets where it will always be available (and knows how to require the workload when appropriate)?